### DISCUSSION TITLE
LLVM flang driver

### DISCUSSION THREAD

**Discussion by awvwgk** [Homebrew/discussions#2753] [Writing Formulae/Casks] [answered]
Should the [flang compiler](http://flang.llvm.org) have a separate formula or would it be preferred to add flang to the enabled projects in the LLVM formula (https://github.com/Homebrew/homebrew-core/blob/5b11df1de24e8f24b320e9356fe92ecabe7110e5/Formula/llvm.rb#L55-L62)?

Found that Macports is building flang together with LLVM ([Portfile](https://github.com/macports/macports-ports/blob/HEAD/lang/llvm-13/Portfile)).

**Comment by carlocab** [comment, marked as answer]
Last I checked Flang was still considered experimental, which is a big part of why it's neither its own formula nor part of the `llvm` formula. (See Homebrew/homebrew-core#65226 and Homebrew/homebrew-core#63081.) The second big part is that it doesn't seem very useful in its current state: it can only parse and unparse Fortran source files, and still needs an external Fortran compiler (e.g. `gfortran`) to compile and link anything.

That said, LLVM do include `flang` in their [binary release](https://github.com/llvm/llvm-project/releases/tag/llvmorg-13.0.0), so I'm not hugely opposed to packaging it in Homebrew/core if it actually is useful despite the limitation described above. I'd be inclined to make it a separate formula since it would need a dependency on `gcc` to be useful.

That said, if you just want to experiment with it, I do have bottles for Catalina and Big Sur (x86_64 only) in my tap: `brew install carlocab/personal/flang`. I've kept it up-to-date, but I don't use it much these days so I can't say anything about how well it works. There's also a Linux bottle, but that was built on Ubuntu 20.04 so you'll need a new-ish `glibc` if you want to use the bottle on Linux.

**Reply by awvwgk** [reply]
I think there is some demand from the Fortran community to use and test flang, independently of its current shape. Unfortunately, there are not yet many package managers which make it easy to test this compiler and provide feedback to the developers.

At least questions about the flang compiler are coming up quite regularly on the Fortran discourse (most recently https://fortran-lang.discourse.group/t/using-flang-in-github-actions/2546).

**Reply by carlocab** [reply]
Ah, whoops. The Linux bottle was a lie. I remember having issues with it initially but eventually working them out. Guess I forgot to publish the bottle that was built in the end.

**Reply by rouson** [reply]
@awvwgk, if I understand correctly, issue #6153  is related to flang and llvm being separate Homebrew packages.  From the above discussion, I assume flang became a separate package because it was considered experimental when the package was created.  I believe flang is  [no longer considered experimental].  Would you be up for adding flang to the enabled projects in the LLVM formula now?  This could alleviate the aforementioned issue if ISO_Fortran_binding.h lands somewhere in clang's default search path for include files.

[no longer considered experimental]: https://blog.llvm.org/posts/2025-03-11-flang-new/
